tests/test_udf.py

```diff
@@ -380,3 +384,49 @@ def apply_udf_data(data: UdfData):
         mehh_path = Path(data["mehh.__file__"])
         assert not mehh_path.exists()
         assert f"Cleaning up temporary UDF deps at {mehh_path.parent}" in caplog.text
+
+    def test_run_udf_on_vector_data_cube_with_logging(self, tmp_path):
+        repository_root = Path(__file__).parent.parent
+        datacube = openeo.DataCube.load_stac(
+            url=str(repository_root / "docker/local_batch_job/example_stac_catalog/collection.json"),
+            temporal_extent=["2023-06-01", "2023-06-09"],
+        )
+
+        geometries = {
+            "type": "Polygon",
+            "coordinates": [
+                [
+                    [5.07, 51.215],
+                    [5.07, 51.22],
+                    [5.08, 51.22],
+                    [5.08, 51.215],
+                    [5.07, 51.215],
+                ]
+            ],
+        }
+
+        vector_cube = datacube.aggregate_spatial(geometries=geometries, reducer="mean")
+
+        udf = """
+from openeo.udf import UdfData
+import logging
+def apply_udf_data(data: UdfData):
+    logging.warn('Hello from UDF!')
+    return data
+"""
+        post_processed = vector_cube.process("run_udf", data=vector_cube, udf=udf, runtime="Python")
+
+        log_dir = os.environ.get("LOG_DIRS", ".").split(",")[0]
+        log_file = Path(log_dir) / "openeo_python.log"
+        open(log_file, "w").close()  # Clear file content
+
+        run_job(
+            {"process_graph": post_processed.flat_graph()},
+            output_file=tmp_path / "out",
+            metadata_file=tmp_path / JOB_METADATA_FILENAME,
+            api_version="2.0.0",
+            job_dir=tmp_path,
+            dependencies=[],
+        )
+        # TODO: Use caplog instead
```

Comment:
@soxofaan Caplog does not seem to take the executor logs. 
The workaround seems more complicated than just reading the logfile:
https://github.com/pytest-dev/pytest/issues/3037#issuecomment-595466044 
Ok to keep like this?

Reply:
sure, if the logging is done from a subprocess, `caplog` won't capture that, so fine to read it straight from file.

However to be correct in this case I think, the UDF is executed in driver process, not executors. But regardless it's still a (driver) subprocess I think

Reply:
The code is called here in the executors: 
https://github.com/Open-EO/openeo-geopyspark-driver/blob/289b6b1e64a31300958c525ba037b2ca39938d2b/openeogeotrellis/backend.py#L1259